test/state_machine_test.rb

```diff
@@ -19,4 +21,99 @@ class Nuntius::StateMachineTest < ActiveSupport::TestCase
     assert_equal "active", user.state
     assert_equal 0, Nuntius::Event.count
   end
+
+  test "firing an multiple event in multiple transaction creates nuntius-events " do
+    user = User.create!(name: "test", email: "test@example.com")
+    Nuntius::Template.create!(description: "template2", klass: "User", transport: "mail", event: "activate", to: "test@example.org", html: "Hello {{name}} - 10 days")
+
+    assert_equal "pending", user.state
+
+    perform_enqueued_jobs(only: [Nuntius::MessengerJob]) do
+      User.transaction do
+        user.activate!
+      end
+      User.transaction do
+        user.disable!
+      end
+      User.transaction do
+        user.activate!
+      end
+      User.transaction do
+        user.save!
+      end
+    end
+    assert_performed_jobs 2
+
+    assert_equal "active", user.state
+    assert_equal 1, Nuntius::Event.count
```

Comment:
Are we really expecting 1? I would've expected 0, as messenger job should clean it up.